### PR TITLE
[frontend/backend] Add Case Rfi and Case Rft Entities (#issue/2977)

### PR DESCRIPTION
**Pull request by Kedae**
Implementation of remaining for RFI/RFT

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfi.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+          <StixCoreObjectLatestHistory stixCoreObjectId={caseRfiData.id} />
+        </Grid>
+      </Grid>
+      <StixCoreObjectOrStixCoreRelationshipNotes
+        stixCoreObjectOrStixCoreRelationshipId={caseRfiData.id}
+        defaultMarking={(caseRfiData.objectMarking?.edges ?? []).map(
+          (edge) => edge.node,
+        )}
+      />
+      <Security needs={[SETTINGS]}>
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfi.tsx]: why Settings ?

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiEditionOverview.tsx [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    readonly name: string;
+  } | null)[]
+  | null;
+  enableReferences?: boolean;
+  handleClose: () => void;
+}
+
+interface CaseRfiEditionFormValues {
+  message?: string
+  createdBy: Option | undefined
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiEditionOverview.tsx]: why not just 'createdBy?: Option' ?

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiKnowledge.js [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+class CaseRfiKnowledgeComponent extends Component {
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiKnowledge.js:115]: component to pure function?

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiPopover.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      <Dialog
+        PaperProps={{ elevation: 1 }}
+        open={displayDelete}
+        keepMounted={true}
+        TransitionComponent={Transition}
+        onClose={handleCloseDelete}
+      >
+        <DialogContent>
+          <DialogContentText>
+            {t('Do you want to delete this request for information case?')}
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiPopover.tsx]: missing translation

**Comment by Archidoit** [comment]
backend build fail because of the migration:
image

origin of the issue:
image
(caseEntitySettings is undefined)

**Comment by Archidoit** [comment]
Filtering not working in Case RFTs and Case RFIs lines: when we add a filter, no line is displayed.

**Comment by Archidoit** [comment]
Exports are not supported

**Comment by Archidoit** [comment]
When we click on the Close cross of the Edition form, nothing happens.
image

**Comment by Archidoit** [comment]
Case RFT/RFI displayed as Unknown in the Knowledge of an entity
image

**Comment by Archidoit** [comment]
In relationships lines, it's the case RFT/RFI id that is displayed (instead of its name)
image

File: opencti-platform/opencti-front/src/private/components/cases/case_rft/CaseRftPopover.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+      <Dialog
+        PaperProps={{ elevation: 1 }}
+        open={displayDelete}
+        keepMounted={true}
+        TransitionComponent={Transition}
+        onClose={handleCloseDelete}
+      >
+        <DialogContent>
+          <DialogContentText>
+            {t('Do you want to delete this request for takedown case?')}
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/cases/case_rft/CaseRftPopover.tsx]: missing translation

**Comment by Archidoit** [comment]
missing translation:
image
(not specific to this issue ;) )

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfi.tsx (hunk 1 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfi.tsx]: It was a mistake, it should be 'KNOWLEDGE_KNUPDATE'

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiPopover.tsx (hunk 4 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiPopover.tsx]: Done ✅

File: opencti-platform/opencti-front/src/private/components/cases/case_rft/CaseRftPopover.tsx (hunk 5 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/cases/case_rft/CaseRftPopover.tsx]: Done ✅

File: opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiEditionOverview.tsx (hunk 2 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/cases/case_rfi/CaseRfiEditionOverview.tsx]: You are right we can write it like that :)

**Comment by Archidoit** [comment]
Issue with the filtering of RFT/RFIs lines
image

**Comment by RomuDeuxfois** [comment]
Case RFT & RFI have same icon

Settings/Case RFT
		- Hidden entity not working ✅
		- External reference not present working on update ✅
		- Missing I18n on mandatory attributes -> Takedown_types

Settings/Case RFI
		- Hidden entity not working ✅
		- External reference not present working on update ✅
		- Missing I18n on mandatory attributes -> Information_types

Case RFT :
Creation
		- Creators still empty ✅
Details
		- Empty field should be '-' ✅
image
		- Linked entities list relationship (maybe global issue)
image
Update
		- Crash sometimes when we removed fields ✅
Delete
               - Miss I18n ✅
image

Case RFI :
Creation
		- Creators still empty ✅
Details
		- Empty field should be '-' ✅
image
Update
		- Crash sometimes when we removed fields ✅
Delete
               - Miss I18n
image ✅

**Comment by RomuDeuxfois** [comment]
Bug founds :
- Redirect not working (on URL) ✅
image
- Missing I18n on mandatory attributes -> Takedown_types
image
- Missing I18n on mandatory attributes -> Information_types
image
- Empty field should be '-' ✅
image
- Miss I18n ✅
image
- Miss I18n ✅
image

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**